packages/platform/types/event-types/event-types_2024_06_14/inputs/update-event-type.input.ts

```diff
@@ -150,77 +166,107 @@ export class UpdateEventTypeInput_2024_06_14 {
   @IsInt()
   @Min(0)
   @IsOptional()
-  @DocsProperty({ description: "Minimum number of minutes before the event that a booking can be made." })
+  @DocsPropertyOptional({
+    description: "Minimum number of minutes before the event that a booking can be made.",
+  })
   minimumBookingNotice?: number;
 
   @IsInt()
   @IsOptional()
-  @DocsProperty({
+  @DocsPropertyOptional({
     description: "Time spaces that can be pre-pended before an event to give more time before it.",
   })
   beforeEventBuffer?: number;
 
   @IsInt()
   @IsOptional()
-  @DocsProperty({
+  @DocsPropertyOptional({
     description: "Time spaces that can be appended after an event to give more time after it.",
   })
   afterEventBuffer?: number;
 
   @IsInt()
   @IsOptional()
-  @DocsProperty({
+  @DocsPropertyOptional({
     description:
       "If you want that this event has different schedule than user's default one you can specify it here.",
   })
   scheduleId?: number;
 
   @IsOptional()
   @ValidateBookingLimistsCount()
-  @DocsProperty({ description: "Limit how many times this event can be booked" })
+  @DocsPropertyOptional({
+    description: "Limit how many times this event can be booked",
+    oneOf: [
+      { $ref: getSchemaPath(BaseBookingLimitsCount_2024_06_14) },
+      { $ref: getSchemaPath(Disabled_2024_06_14) },
+    ],
+  })
+  @Type(() => Object)
   bookingLimitsCount?: BookingLimitsCount_2024_06_14;
 
   @IsOptional()
   @IsBoolean()
-  @DocsProperty({
+  @DocsPropertyOptional({
     description:
       "This will limit your availability for this event type to one slot per day, scheduled at the earliest available time.",
   })
   onlyShowFirstAvailableSlot?: boolean;
 
   @IsOptional()
   @ValidateBookingLimistsDuration()
-  @DocsProperty({ description: "Limit total amount of time that this event can be booked" })
+  @DocsPropertyOptional({
+    description: "Limit total amount of time that this event can be booked",
+    oneOf: [
+      { $ref: getSchemaPath(BaseBookingLimitsDuration_2024_06_14) },
+      { $ref: getSchemaPath(Disabled_2024_06_14) },
+    ],
+  })
+  @Type(() => Object)
   bookingLimitsDuration?: BookingLimitsDuration_2024_06_14;
 
   @IsOptional()
   @ValidateBookingWindow()
-  @DocsProperty({
+  @DocsPropertyOptional({
     description: "Limit how far in the future this event can be booked",
     oneOf: [
       { $ref: getSchemaPath(BusinessDaysWindow_2024_06_14) },
       { $ref: getSchemaPath(CalendarDaysWindow_2024_06_14) },
       { $ref: getSchemaPath(RangeWindow_2024_06_14) },
```

Comment:
add `Disabled_2024_06_14` as well
same for `create-event-type.input.ts`

Reply:
Well spotted! Implemented in [refactor: add disabled option to booking window](https://github.com/calcom/cal.com/pull/16854/commits/75fb06324893893889e6e3e7f0d98335e195099d)